tests/testthat/test-method-differentialExpression.R

```diff
@@ -0,0 +1,546 @@
+# Tests for differential expression methods
+
+test_that('differentialExpression returns a correctly formatted data.table', {
+
+  testCollection <- testCountDataCollection
+  testSampleMetadata <- getSampleMetadata(testCollection)
+
+
+  # A Binary comparator variable
+  comparatorVariable <- veupathUtils::Comparator(
+                          variable = veupathUtils::VariableMetadata(
+                            variableSpec = VariableSpec(
+                              variableId = 'binA',
+                              entityId = 'entity'
+                            ),
+                            dataShape = veupathUtils::DataShape(value="BINARY")
+                          ),
+                          groupA = veupathUtils::BinList(
+                            S4Vectors::SimpleList(
+                              c(veupathUtils::Bin(
+                                binLabel="binA_a"
+                              ))
+                            )
+                          ),
+                          groupB = veupathUtils::BinList(
+                            S4Vectors::SimpleList(
+                              c(veupathUtils::Bin(
+                                binLabel="binA_b"
+                              ))
+                            )
+                          )
+  )
+
+  result <- differentialExpression(testCollection, comparator=comparatorVariable, method='DESeq', verbose=F)
+  expect_equal(length(result@droppedColumns), 0)
+  dt <- result@data
```

Comment:
> So this is a 288 row table with one column - SampleID.

Yup!

> Is this useful for anything? If it had a second column with groupA/B membership, maybe it would?

It's not being used right now, but it's supposed to help us tell the user which samples get thrown out in the computation because of bad data or whatever the reason. I like the idea of adding the groupA/groupB membership there. Not sure yet where it would be used, but when the need comes I'm glad to know we. have a natural place to add that information.